numba/cuda/cudadrv/driver.py

```diff
@@ -775,8 +775,7 @@ def _module_finalizer(context, handle):
 
     def core():
         def cleanup():
-            if modules:
-                del modules[handle.value]
+            assert handle.value not in modules
```

Comment:
Can you add a comment explaining why this is the case? Also, why is the call to `cuModuleUnload()` made here rather than in the module's finalizer?